cs1/graphics.py

```diff
@@ -29,12 +30,16 @@
 
 _DRAW_BORDER = True
 
+_tb = tokenbucket.TokenBucket(1000, 1)
+
 # Global functionality
 
 def _check():
   global _canvas
   if _canvas == None:
     raise RuntimeError("Canvas is not open yet.")
+  else:
```

Comment:
This seems orthogonal to the purpose of this function, especially because this function is called in every function that expects the canvas to be open, not necessarily every function that has to be rate-limited.

I would suggest creating a [decorator](https://towardsdatascience.com/the-simplest-tutorial-for-python-decorator-dadbf8f20b0f) and then using that on all functions that should be rate limited. For example:

```
def rate_limited(f):
  def maybe_limit():
    ...
  return maybe_limit

@rate_limited
def draw_blah(...):
  ...
```